metrics/sources/summary/summary.go

```diff
@@ -194,10 +194,30 @@ func (this *summaryMetricsSource) decodePodStats(metrics map[string]*MetricSet,
 
 	for _, container := range pod.Containers {
 		key := PodContainerKey(ref.Namespace, ref.Name, container.Name)
+		// This check ensures that we are not replacing metrics of running container with metrics of terminated one if
+		// there are two exactly same containers reported by kubelet.
+		if _, exist := metrics[key]; exist {
+			glog.V(8).Infof("Metrics reported from two containers with the same key: %v. Create time of "+
+				"containers are %v and %v. Metrics from the older container are going to be dropped.", key,
+				container.StartTime.Time, metrics[key].CreateTime)
+			if containerIsTerminated(&container, metrics[key].CreateTime) {
+				continue
+			}
+		}
 		metrics[key] = this.decodeContainerStats(podMetrics.Labels, &container, false)
 	}
 }
 
+func containerIsTerminated(container *stats.ContainerStats, otherStartTime time.Time) bool {
+	if container.StartTime.Time.Before(otherStartTime) {
+		if *container.CPU.UsageNanoCores == 0 {
```

Comment:
Thanks for the comment, it really makes sense. I think checking of CPU and Memory usage should be good enough heuristic.

I have updated a code and test.